src/Faker/Provider/en_ZA/Person.php

```diff
@@ -148,7 +148,7 @@ public function idNumber(\DateTime $birthdate = null, $citizen = true, $gender =
         }
         $birthDateString = $birthdate->format('ymd');
 
-        switch (strtolower($gender)) {
+        switch (strtolower($gender ?: '')) {
```

Comment:
`$gender` is `null` by default, but in php 8.1 it's deprecated to use `null` as parameter for `strtolower`.
```
  1x: strtolower(): Passing null to parameter #1 ($string) of type string is deprecated
    1x in PersonTest::testIdNumberWithDefaults from Faker\Test\Provider\en_ZA
```

Reply:
Thanks!